modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/ImageMosaicCogOnlineTest.java

```diff
@@ -26,34 +28,51 @@
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
 import java.util.Properties;
 import junit.framework.TestCase;
 import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.geotools.coverage.grid.GridCoverage2D;
 import org.geotools.coverage.grid.GridEnvelope2D;
 import org.geotools.coverage.grid.GridGeometry2D;
 import org.geotools.coverage.grid.io.AbstractGridCoverage2DReader;
 import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.coverage.grid.io.GranuleSource;
+import org.geotools.coverage.grid.io.HarvestedSource;
+import org.geotools.data.Query;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.geotools.gce.imagemosaic.catalog.GranuleCatalog;
 import org.geotools.geometry.GeneralEnvelope;
 import org.geotools.test.TestData;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Test;
+import org.opengis.feature.simple.SimpleFeature;
 import org.opengis.parameter.GeneralParameterValue;
 import org.opengis.parameter.ParameterValue;
 
 /** Testing using COG remote granules on an ImageMosaic */
 public class ImageMosaicCogOnlineTest extends TestCase {
```

Comment:
So are these tests going to run in some builds? PR checks? build server? Or just locally if manually enabled?

Reply:
Also, I see code trying to handle a list of URLs during harvesting (one of the harvested resources). Is there a test for that case?

Reply:
Yep. there is the online test checking it. (which was failing, bytheway. Fixed it)
I have also updated the timeMosaic test to harvest a List of 2 URLs instead of just a single one.